proposals/texture-buffers.md

```diff
@@ -0,0 +1,405 @@
+
+
+## Feature Index
+
+#### `"texture-buffers"`
+
+Allows the use of the `texture_buffer` type in WGSL, and the creation of `GPUTextureBufferView` objects.
+
+This feature adds the following optional API surfaces:
+
+- New WGSL extensions:
+  - `texture_buffers`
+
+
+## Plain color formats
+
+[[Add new column to format table for `TEXTURE_BUFFER`]]
+
+| Format                    | `TEXTURE_BUFFER` |
+| ------------------------- | ---------------- |
+| **8-bit per component**   |
+| `r8unorm`                 |                  |
+| `r8snorm`                 |                  |
+| `r8uint`                  | ✔                |
+| `r8sint`                  | ✔                |
+| `rg8unorm`                |                  |
+| `rg8snorm`                |                  |
+| `rg8uint`                 |                  |
+| `rg8sint`                 |                  |
+| `rgba8unorm`              | ✔                |
+| `rgba8unorm-srgb`         |                  |
+| `rgba8snorm`              |                  |
+| `rgba8uint`               | ✔                |
+| `rgba8sint`               | ✔                |
+| `bgra8unorm`              |                  |
+| `bgra8unorm-srgb`         |                  |
+| **16-bit per component**  |
+| `r16uint`                 | ✔                |
+| `r16sint`                 | ✔                |
+| `r16float`                | ✔                |
+| `rg16uint`                |                  |
+| `rg16sint`                |                  |
+| `rg16float`               |                  |
+| `rgba16uint`              | ✔                |
+| `rgba16sint`              | ✔                |
+| `rgba16float`             | ✔                |
+| **32-bit per component**  |
+| `r32uint`                 | ✔                |
+| `r32sint`                 | ✔                |
+| `r32float`                | ✔                |
+| `rg32uint`                |                  |
+| `rg32sint`                |                  |
+| `rg32float`               |                  |
+| `rgba32uint`              | ✔                |
+| `rgba32sint`              | ✔                |
+| `rgba32float`             | ✔                |
+| **mixed component width** |
+| `rgb10a2uint`             |                  |
+| `rgb10a2unorm`            |                  |
+| `rg11b10ufloat`           |                  |
+
+
+# Appendix A: Implementation details
+
+
+### Vulkan
+
+In Vulkan, a `read_write` texture buffer would map to a storage texel buffer decorated as `Coherent`, and shader accesses would be performed with the `OpImageRead` and `OpImageWrite` instructions.
+A texture buffer with a `read-only` access mode could use a uniform texel buffer instead, which would use `OpImageFetch` instead `OpImageRead`.
+
+When the `TEXTURE_BUFFER` usage flag is set on buffer creation, both of the Vulkan texel buffer bits would be set:<br/>
+`VK_BUFFER_USAGE_UNIFORM_TEXEL_BUFFER_BIT | VK_BUFFER_USAGE_STORAGE_TEXEL_BUFFER_BIT`
```

Comment:
Why is `VK_BUFFER_USAGE_UNIFORM_TEXEL_BUFFER_BIT` needed?

Reply:
It'd be needed if the implementation wanted to use a uniform texel buffer for the read-only case, instead of using a storage texel buffer.